backend/src/controllers/__tests__/CMSTutorialController.unit.test.ts

```diff
@@ -0,0 +1,202 @@
+import Server from "../../App";
+import supertest from "supertest";
+import { MongoClient, Db, Collection } from "mongodb";
+import { BAD_REQUEST_TEXT, BAD_OBJECTID_PARSE_TEXT } from "../../constants";
+import PartialTutorial from "../../dtos/PartialTutorial.dto";
+import Tutorial from "../../dtos/Tutorial.dto";
+
+/**
+ * This test utilises [supertest]{@link https://www.npmjs.com/package/supertest}
+ * and initialises the full server to perform the test per test.
+ *
+ * @author ale8k
+ */
+describe("CMSTutorialController", () => {
+    let connection: MongoClient;
+    const usersCollectionName = "cms-users";
+    const tutorialsCollectionName = "tutorials";
+    let db: Db;
+    let usersCollection: Collection;
+    let tutorialsCollection: Collection;
+
+    beforeAll(async () => {
+        process.env.PORT = "3003";
+        process.env.DB_URL = global.__MONGO_URI__;
+        process.env.DB_NAME = global.__MONGO_DB_NAME__;
+        process.env.BCRYPT_SALT = "12";
+        process.env.JWT_KEY = "iamakeylol";
+        process.env.JWT_EXPIRY = "2 days";
+
```

Comment:
if these are the configs that you're gonna go with why not set a `test-configs.ts` file.

Reply:
They're being manually set at test run-time for this we'll need another lib to set jest-environment. I'll add it in

Reply:
I'll shove them in an exporting function for now, think that's probably better lol.